src/lib/types.h

```diff
@@ -445,19 +699,24 @@ typedef struct pgp_revoke_t {
 
 typedef struct pgp_user_prefs_t {
     // preferred symmetric algs (pgp_symm_alg_t)
-    uint8_t *symm_algs;
-    size_t   symm_alg_count;
+    std::vector<uint8_t> symm_algs{};
     // preferred hash algs (pgp_hash_alg_t)
-    uint8_t *hash_algs;
-    size_t   hash_alg_count;
+    std::vector<uint8_t> hash_algs{};
     // preferred compression algs (pgp_compression_type_t)
-    uint8_t *z_algs;
-    size_t   z_alg_count;
+    std::vector<uint8_t> z_algs{};
     // key server preferences (pgp_key_server_prefs_t)
-    uint8_t *ks_prefs;
-    size_t   ks_pref_count;
+    std::vector<uint8_t> ks_prefs{};
     // preferred key server
-    uint8_t *key_server;
+    std::string key_server{};
+
+    void set_symm_algs(const uint8_t algs[], size_t len);
```

Comment:
Why don't we add `set_xxx_algs` overloads that take `vector` as parameters? It would simplify other parts of code.

Reply:
@rrrooommmaaa Hm, actually we don't require uint8_t algs[] here at all. Will update the PR.